personalize.admin.inc

```diff
@@ -85,6 +85,89 @@ function personalize_admin_form_submit($form, &$form_state) {
 }
 
 /**
+ * Form for changing the status of an agent.
+ *
+ * @param $agent_name
+ *   THe name of the agent the form is for.
+ * @param array $agent_status
+ *   The current status of the agent.
+ * @return array
+ *   Array representing the status change form.
+ */
+function personalize_status_change_form($form, &$form_state, $agent_name, $agent_status) {
+  $friendly_statuses = personalize_get_agent_status_map();
+  $form = array();
+  $form['agent_name'] = array(
+    '#type' => 'value',
+    '#value' => $agent_name,
+  );
+  switch($agent_status) {
```

Comment:
Hmm, then the personalize_allowed_status_transitions would have to know about things like button text. I wanted to keep that one just as the canonical statement of which statuses can be transitioned to from each status, whereas here it's about presenting an actual UI. Per my comment above, there's definitely some clean-up to be done here but I'll leave it is as a todo for now (I wonder when we'll actually get time to do our todos :P)